todo_app/app.py

```diff
@@ -1,23 +1,80 @@
+import os
 from flask import Flask, redirect,request,render_template
+from flask_login import LoginManager, login_required, login_user
+import requests
 from todo_app.data.todo_items import get_items, add_item, update_item
+from todo_app.data.user import User
 from todo_app.flask_config import Config
+
 def create_app():
     app = Flask(__name__)
     app.config.from_object(Config())
 
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        client_id = os.getenv('GITHUB_CLIENT_ID')
```

Comment:
It would be good to mention this ID & the `GITHUB_CLIENT_SECRET` somewhere so users know they need to set them - the easiest place being just adding the keys (but not the values!) to your `.env.template` file
(If this were a real project, adding some advice for how new developers are expected to set up those would also be really important!)